qabelbox/src/main/java/de/qabel/qabelbox/storage/AbstractNavigation.java

```diff
@@ -108,13 +110,23 @@ public String getName() {
         return filepath.substring(filepath.lastIndexOf('/') + 1, filepath.length());
     }
 
-    protected File blockingDownload(String prefix, String name, BoxTransferListener boxTransferListener) throws QblStorageNotFound {
+    protected File blockingDownload(String prefix, String name, BoxTransferListener boxTransferListener) throws QblStorageException {
         File file = transferManager.createTempFile();
         int id = transferManager.download(prefix, name, file, boxTransferListener);
         if (transferManager.waitFor(id)) {
             return file;
         } else {
-            throw new QblStorageNotFound("File not found. Prefix: " + prefix + " Name: " + name);
+            try {
+                Exception ex = transferManager.lookupError(id);
```

Comment:
No need for the variable ex, you either just inspect the exception directly or throw it directly.